shared/js/background/tab-manager.es6.js

```diff
@@ -121,6 +137,99 @@ class TabManager {
                 tab.updateSite(request.url)
             }
         }
+
+        this.saveStateToStorage()
+    }
+
+    // if the background page is not persistent (service worker) we need to try to recover
+    // TabManager state from storage
```

Comment:
We are serializing and deserializing because service worker can be killed by the browser when inactive and we don't want to lose tab state - grade, trackers loaded, etc